### PR TITLE
Schedule events

### PR DESCRIPTION
**Pull request by joakim-hove**
[ This is mainly put up as a request for comments - not yet merge ready. ]

With this PR we have created an initial system to record events from the Schedule file. The purpose of this is that you can query the events table for specific events, and then decide whether a potentially expensive calculation is required.

This is absolutely work in progress - what I would very much like feedback on are:

 1. Which events are required - the current list in Events.hpp is somewhat random; and I will *not* guarantee that they are set correctly.
 2. Querying for events is based on implementing `Events::operator()` - makes sense?

### CODE REVIEW

File: opm/parser/eclipse/EclipseState/Schedule/Events.hpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+#include <cstdint>
+#include <vector>
+
+namespace Opm
+{
+    namespace ScheduleEvents {
+        // These values are used as bitmask - 2^n structure is essential.
+        enum Events {
+            NEW_WELL = 1,
```
Comment (atgeirr) [inline opm/parser/eclipse/EclipseState/Schedule/Events.hpp]: Is there a reason for skipping 2 in the sequence of values?

File: opm/parser/eclipse/EclipseState/Schedule/Events.hpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+           ...
+
+      Querying events is done with operator().
+    */
+
+    class Events {
+    public:
+        Events();
+        void addEvent(ScheduleEvents::Events event, size_t reportStep);
+        bool operator()(ScheduleEvents::Events event, size_t reportStep) const;
```
Comment (atgeirr) [inline opm/parser/eclipse/EclipseState/Schedule/Events.hpp]: While somewhat cool, I think that operator() is more likely to confuse users than to inform them. I think a more (boring) traditional hasEvent() or stepHasEvent() method name would be better.

**Comment by atgeirr** [comment]
This looks good, and quite easy to use from client code. I have not by any means checked that all events that should be added are actually added, but the parts I looked at seemed ok.

File: opm/parser/eclipse/EclipseState/Schedule/Events.hpp (hunk 2 above)
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Schedule/Events.hpp]: Thanks for looking; I also thought the `operator()` was kind of cool - but of course coolness is not the metric our code should be measured by. Will update.

**Comment by joakim-hove** [comment]
>I have not by any means checked that all events that should be added are actually added

There are two aspects to this:

1. Enumerate the list of events we are interested in.
2. Actually record the events we are interested in.

There is a trade off here where fine grained events can lead to fine grained decisions in the downstream simulator code - which is good; and also potentially increase the complexity in the event recording code - which is bad. So we should not introduce fine grained events for the sake of coolness - but rather limit ourselves to the events which we assume will differentiate on in the downstream code. We can discuss this further.

**Comment by joakim-hove** [comment]
Fixup using `hasEvent()` instead of `operator()` pushed.

**Pull request merged**